dhctl/pkg/system/node/gossh/file.go

```diff
@@ -0,0 +1,250 @@
+		}
+	}()
+
+	err = os.WriteFile(srcPath, data, 0o600)
+	if err != nil {
+		return fmt.Errorf("write data to tmp file: %w", err)
+	}
+
+	err = f.Upload(ctx, srcPath, remotePath)
+	return err
+}
+
+func (f *SSHFile) Download(ctx context.Context, remotePath, dstPath string) error {
+	fType, err := getRemoteFileStat(f.sshClient, remotePath)
+	if err != nil {
+		return err
+	}
+
+	if fType != "directory" {
+		// regular file logic
+		scpClient, err := scp.NewClientBySSH(f.sshClient)
+		if err != nil {
+			return err
+		}
+		defer scpClient.Close()
+		localFile, err := os.Create(dstPath)
+		if err != nil {
+			return fmt.Errorf("failed to open local file: %w", err)
+		}
+		defer localFile.Close()
+		if err := scpClient.CopyFromRemote(ctx, localFile, remotePath); err != nil {
+			return fmt.Errorf("failed to copy file to remote host: %w", err)
+		}
+	} else {
+		// recursive copy logic
+		filesString, err := getRemoteFilesList(f.sshClient, remotePath)
+		if err != nil {
+			return err
+		}
+
+		if filepath.Base(dstPath) != filepath.Base(remotePath) {
+			dstPath = dstPath + "/" + filepath.Base(remotePath)
+		}
+
+		err = os.MkdirAll(dstPath, os.ModePerm)
+		if err != nil {
+			return err
+		}
+
+		re := regexp.MustCompile(`\s+`)
+		files := re.Split(filesString, -1)
+		for _, file := range files {
+			f.Download(ctx, remotePath+"/"+file, dstPath+"/"+file)
+		}
+	}
+
+	return nil
+}
+
+// Download remote file and returns its content as an array of bytes.
+func (f *SSHFile) DownloadBytes(ctx context.Context, remotePath string) ([]byte, error) {
+	dstPath, err := CreateEmptyTmpFile()
+	if err != nil {
+		return nil, fmt.Errorf("create target tmp file: %v", err)
+	}
+	defer func() {
+		err := os.Remove(dstPath)
+		if err != nil {
+			log.InfoF("Error: cannot remove tmp file '%s': %v\n", dstPath, err)
+		}
+	}()
+
+	err = f.Download(ctx, remotePath, dstPath)
+	if err != nil {
+		return nil, fmt.Errorf("download target tmp file: %v", err)
+	}
+
+	data, err := os.ReadFile(dstPath)
+	if err != nil {
+		return nil, fmt.Errorf("reading tmp file '%s': %w", dstPath, err)
+	}
+
+	return data, nil
+}
+
+func getRemoteFileStat(client *ssh.Client, remoteFilePath string) (string, error) {
+	session, err := client.NewSession()
+	if err != nil {
+		return "", fmt.Errorf("failed to create session: %w", err)
+	}
+	defer session.Close()
+
+	command := fmt.Sprint("stat -c %F " + remoteFilePath)
+	output, err := session.CombinedOutput(command)
+
+	return strings.TrimSpace(string(output)), err
+}
+
+func getRemoteFilesList(client *ssh.Client, remoteFilePath string) (string, error) {
+	session, err := client.NewSession()
+	if err != nil {
+		return "", fmt.Errorf("failed to create session: %w", err)
+	}
+	defer session.Close()
+
+	command := fmt.Sprint("ls " + remoteFilePath)
+	output, err := session.CombinedOutput(command)
+
+	return strings.TrimSpace(string(output)), err
+}
+
+func CreateEmptyTmpFile() (string, error) {
+	tmpPath := filepath.Join(
+		os.TempDir(),
```

Comment:
we have temp dir parameter https://github.com/deckhouse/deckhouse/blob/dc47d38147ed6c0806d14c6d8ef039540320bedd/dhctl/pkg/app/app.go#L44-L43
I think better use this dir for all